server/controllers/userController.js

```diff
@@ -1,107 +1,117 @@
-import dotenv from 'dotenv';
 import jwt from 'jsonwebtoken';
 import moment from 'moment';
 import db from '../models';
 
-dotenv.config({ silent: true });
 
 const secret = process.env.SECRET || 'Happypeopledontkeepsecret';
 const expires = moment().add(1, 'days').valueOf();
 
 
 const UserController = {
-  createNewUser(req, res) {
-    db.users.findOne({ where: { email: req.body.email } })
+  createNewUser(request, response) {
+    const firstName = request.body.firstName;
+    const lastName = request.body.lastName;
+    const email = request.body.email;
+    const userName = request.body.userName;
+    const password = request.body.password;
+    if (!firstName || !lastName || !email || !userName || !password) {
+      return response.status(400).send({
+        message: 'The paramaters are incomplete',
+      });
+    }
+    db.users.findOne({ where: { $or: { email, userName } } })
       .then((userExists) => {
         if (userExists) {
-          return res.status(400).send({
-            message: 'There is a user already existing with this email'
+          return response.status(400).send({
+            message: `There is a user already existing
+            with this email or userName`
           });
         }
-        db.users
-          .create(req.body)
+        return db.users
+          .create(request.body)
           .then((user) => {
             const payload = {
               userId: user.id,
               roleId: user.roleId
             };
             const token = jwt.sign(payload, secret, { expiresIn: expires });
-            return res.status(201).send({
+            return response.status(201).send({
               message: 'User was successfully created',
               token,
               data: user
             });
           })
           .catch((err) => {
-            res.status(400).send({
-              message: err.message
+            response.status(400).send({
+              message: 'There was a problem creating this user', err
             });
           });
       });
   },
 
-  loginUser(req, res) {
+  loginUser(request, response) {
     db.users
       .findOne({
         where: {
-          email: req.body.email
+          email: request.body.email
         }
       })
       .then((user) => {
-        if (user && user.validatePassword(req.body.password)) {
+        if (!user) {
+          return response.status(400).send({
```

Comment:
Due to security reasons, it's not ideal to send the particular field that is wrong to a user during Authentication.